File: src/main/java/br/com/alura/forum/model/PossibleSpammer.java
package br.com.alura.forum.model;

import java.time.Duration;
import java.time.Instant;
import java.util.List;

import br.com.alura.forum.model.topic.domain.Topic;

public class PossibleSpammer {

    private List<Topic> topics;

    public PossibleSpammer(List<Topic> topics) {
        this.topics = topics;
    }

    public boolean hasTopicLimitExceeded() {
        return this.topics.size() >= 4;
    }

    public long minutesToNextTopic(Instant from) {
        Instant instantOfTheOldestTopic = topics.get(0).getCreationInstant();

        return Duration.between(from, instantOfTheOldestTopic)
                .getSeconds() / 60;
    }
}

File: src/main/kotlin/br/com/alura/forum/model/PossibleSpammer.kt
package br.com.alura.forum.model

import br.com.alura.forum.model.topic.domain.Topic
import java.time.Duration
import java.time.Instant

class PossibleSpammer(private val topics: List<Topic>) {
    val hasTopicLimitExceeded = this.topics.size >= 4

    fun minutesToNextTopic(from: Instant): Long {
        val instantOfTheOldestTopic = this.topics[0].creationInstant

        return Duration.between(from, instantOfTheOldestTopic).seconds/60
    }
}